lib/components/narrative/metro/metro-itinerary.tsx

```diff
@@ -400,54 +431,64 @@ class MetroItinerary extends NarrativeItinerary {
+                        isContinuousDropoff={isContinuousDropoff}
+                        phoneNumber={phone}
+                        shrink={false}
+                        textOnly
                       />
-                    )
-                  )}
-                </SecondaryInfo>
-                <SecondaryInfo>
-                  {transitFare === null || transitFare < 0 ? (
-                    <FormattedMessage id="common.itineraryDescriptions.noTransitFareProvided" />
-                  ) : (
-                    // TODO: re-implement TNC fares for metro UI?
-                    <FormattedNumber
-                      currency={fareCurrency}
-                      currencyDisplay="narrowSymbol"
-                      // This isn't a "real" style prop
-                      // eslint-disable-next-line react/style-prop-object
-                      style="currency"
-                      value={transitFare / 100}
+                    ) : (
+                      firstTransitStop && (
+                        <FormattedMessage
+                          id="components.MetroUI.fromStop"
+                          values={{ stop: firstTransitStop }}
+                        />
+                      )
+                    )}
+                  </SecondaryInfo>
+                  <SecondaryInfo>
+                    {transitFare === null || transitFare < 0 ? (
+                      <FormattedMessage id="common.itineraryDescriptions.noTransitFareProvided" />
+                    ) : (
+                      // TODO: re-implement TNC fares for metro UI?
+                      <FormattedNumber
+                        currency={fareCurrency}
+                        currencyDisplay="narrowSymbol"
+                        // This isn't a "real" style prop
+                        // eslint-disable-next-line react/style-prop-object
+                        style="currency"
+                        value={transitFare / 100}
+                      />
+                    )}
+                  </SecondaryInfo>
+                  <SecondaryInfo>
+                    <FormattedMessage
+                      id="components.MetroUI.timeWalking"
+                      values={{
+                        time: (
+                          <FormattedDuration duration={itinerary.walkTime} />
+                        )
+                      }}
                     />
-                  )}
-                </SecondaryInfo>
-                <SecondaryInfo>
-                  <FormattedMessage
-                    id="components.MetroUI.timeWalking"
-                    values={{
-                      time: <FormattedDuration duration={itinerary.walkTime} />
-                    }}
-                  />
-                </SecondaryInfo>
+                  </SecondaryInfo>
+                </ItineraryDetails>
                 <DepartureTimes>
                   <FormattedMessage id="components.MetroUI.leaveAt" />{' '}
                   <DepartureTimesList
```

Comment:
Ahh I agree this feels messy in the expanded itin... unfortunately moving this button (sometimes list of buttons) into the `ItineraryDetails` list works great for screen readers but really otherwise breaks the UI 😅 Maybe there's a different way to separate it other than adding to the `ItineraryDetails` list? or maybe i'm missing something?

Reply:
Great idea, thank you!